pkg/provisioner/ironic/ironic.go

```diff
@@ -1613,6 +1617,33 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		_, err := p.hardPowerOff()
+		if err != nil {
+			result.Dirty = true
+			switch err.(type) {
+			case HostLockedError:
+				p.log.Info("could not power off host, busy")
+				return retryAfterDelay(powerRequeueDelay)
```

Comment:
The `retryAfterDelay` is currently equivalent to an operation continue. Is it ok to manage this case as not an error that could increase the error count (see below)?

Reply:
Yes, my understanding is that we would not want to classify the scenario where the node is busy and cannot currently be powered off as an error.

Reply:
Fixed.